packages/components/src/mobile/keyboard-aware-flat-list/index.ios.js

```diff
@@ -42,70 +44,137 @@ export const KeyboardAwareFlatList = ( {
 		},
 	} );
 
+	useEffect( () => {
+		let willShowSubscription;
+		let showSubscription;
+		let hideSubscription;
+
+		if ( scrollEnabled ) {
+			willShowSubscription = Keyboard.addListener(
+				'keyboardWillShow',
+				() => {
+					isKeyboardVisible.current = true;
+				}
+			);
+			showSubscription = Keyboard.addListener(
+				'keyboardDidShow',
+				( { endCoordinates } ) => {
+					setKeyboardSpace( endCoordinates.height );
+				}
+			);
+			hideSubscription = Keyboard.addListener( 'keyboardWillHide', () => {
+				if ( ! RCTAztecView.InputState.isFocused() ) {
+					setKeyboardSpace( 0 );
+				}
+				isKeyboardVisible.current = false;
+			} );
+		}
+		return () => {
+			willShowSubscription?.remove();
+			showSubscription?.remove();
+			hideSubscription?.remove();
+		};
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [] );
+
+	const onScrollToInput = useCallback( () => {
+		if (
+			! isEditingText.current ||
+			! scrollEnabled ||
+			( isKeyboardVisible.current && keyboardSpace === 0 )
+		) {
+			return;
+		}
+
+		const textInput = RCTAztecView.InputState.getCurrentFocusedElement();
+		if ( textInput ) {
+			textInput.measureInWindow( ( _x, y, _width, height ) => {
+				const caretYPosition =
+					currentCaretYCoordinate.current || height;
+				const textInputOffset = y + caretYPosition;
+
+				const offset =
+					latestContentOffsetY.value +
+					( textInputOffset - screenOffset );
+
+				if (
+					listRef.current &&
+					offset > 0 &&
+					textInputOffset > screenOffset
+				) {
+					listRef.current.scrollToOffset( {
+						x: 0,
+						offset,
+						animated: true,
+					} );
+				}
+			} );
+		}
+	}, [
+		scrollEnabled,
+		isEditingText,
+		screenOffset,
+		keyboardSpace,
+		latestContentOffsetY,
+	] );
+
+	useEffect( () => {
+		if ( keyboardSpace !== 0 ) {
+			onScrollToInput();
+		}
+	}, [ keyboardSpace, onScrollToInput ] );
+
+	const onCaretChange = useCallback(
+		( { caretY } ) => {
+			const isFocused =
+				!! RCTAztecView.InputState.getCurrentFocusedElement();
+			isEditingText.current = isFocused;
+
+			if ( ! isFocused ) {
+				return;
+			}
+
+			currentCaretYCoordinate.current = caretY;
+			onScrollToInput();
+		},
+		[ onScrollToInput ]
+	);
+
+	useEffect( () => {
+		if ( scrollEnabled ) {
+			RCTAztecView.InputState.addCaretChangeListener( onCaretChange );
+		}
```

Comment:
If `scrollEnabled` is set to `false`, the caret change listener won't be removed upon component unmount. Hence, we could consider removing it in this case:
```suggestion
		if ( scrollEnabled ) {
			RCTAztecView.InputState.addCaretChangeListener( onCaretChange );
		} else {
			RCTAztecView.InputState.removeCaretChangeListener( onCaretChange );
		}
```